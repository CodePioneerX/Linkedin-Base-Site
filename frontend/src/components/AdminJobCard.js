import React, { useState, useEffect } from "react";
import { useSelector } from "react-redux";
import { useNavigate } from 'react-router-dom';
import axios from "axios";
import { Container, Row, Col, Dropdown, DropdownButton } from 'react-bootstrap';
import { Link } from "react-router-dom";
import { faGripVertical } from "@fortawesome/free-solid-svg-icons";
import '../Assets/css/Admin.css';

const AdminJobCard = (props) => {

  const navigate = useNavigate();

  // Define state variables
  const [reportMessages, setReportMessages] = useState("");
  
  const userLogin = useSelector((state) => state.userLogin);
  const { userInfo } = userLogin;

  const updateToken = useSelector((state) => state.updateToken);
  const { loading, auth } = updateToken;

  const banHandler = async () => {
    const config = {
      headers: {
          'Content-type': 'application/json',
          Authorization: `Bearer ${auth.access}`   
      }
    }
    const response = await axios.put(`http://insightwearai.sytes.net:8000/api/users/ban/${author_id}`, {}, config)
    window.location.reload()
  }

  const dismissReportHandler = async () => {
    const config = {
      headers: {
          'Content-type': 'application/json',
          Authorization: `Bearer ${auth.access}`   
      }
    }
    const response = await axios.delete(`http://insightwearai.sytes.net:8000/api/jobs/report/dismiss/${id}`, config)
    window.location.reload()
  }

  const deleteJobHandler = async () => {
    const response = await axios.delete(`http://insightwearai.sytes.net:8000/api/job/delete/${job_id}`)
    window.location.reload()
  }

  const viewAuthor = () => {
    navigate('/profileScreen', { state: {data: author_id} })
  }

  const viewSender = () => {
    navigate('/profileScreen', { state: {data: sender_id} })
  }

  const viewJob = () => {
    navigate('/job', { state: {job_id: job_id} })
  }

  const { id, sender_id, sender_name, sender_email, job_id, job_title, job_company, job_location, message, author_id, author_name } = props.report

  return (
    <>   
      <Container className="adminCardContainer">
        <div className="adminCardBody">
          <Row>
            <Col xs={8} md={10}>
              <Row className="adminCardRow">
                <h4>Job Author: {author_name}</h4>
              </Row>
              <Row className="adminCardRow">
                <h4>{job_title} at {job_company} in {job_location}</h4>
              </Row>
              <Row className="adminCardRow">
                  <p><strong>{sender_name}</strong> ({sender_email}): "{message}"</p>
              </Row>
              <Row className="adminCardRow">
                <hr style={{width: "100%"}}/>
              </Row>
            </Col>
            <Col xs={4} md={2} style={{display:'flex', justifyContent:'right'}}>
              <Row>
                <DropdownButton id="dropdownButton" title=''>
                  <Dropdown.Item as="button" onClick={viewAuthor}>View Author</Dropdown.Item>
                  <Dropdown.Item as="button" onClick={viewSender}>View Sender</Dropdown.Item>
                  <Dropdown.Item as="button" onClick={viewJob}>View Job</Dropdown.Item>
                  <Dropdown.Item as="button" onClick={dismissReportHandler}>Dismiss Report</Dropdown.Item>
                  <Dropdown.Item as="button" onClick={deleteJobHandler}>Delete Job</Dropdown.Item>
                  <Dropdown.Item as="button" onClick={banHandler}>Ban User</Dropdown.Item>
                </DropdownButton>
              </Row>
            </Col>
          </Row>
        </div>
      </Container>
    </>
  );
};

export default AdminJobCard;
