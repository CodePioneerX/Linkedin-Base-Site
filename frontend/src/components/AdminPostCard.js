import React, { useState, useEffect } from "react";
import { useSelector } from "react-redux";
import { useNavigate } from 'react-router-dom';
import axios from "axios";
import { Container, Row, Col, Dropdown, DropdownButton } from 'react-bootstrap';
import { Link } from "react-router-dom";
import { faGripVertical } from "@fortawesome/free-solid-svg-icons";
import '../Assets/css/Admin.css';

const AdminPostCard = (props) => {
  
  const navigate = useNavigate();

  // Define state variables
  const [reportMessages, setReportMessages] = useState("");

  const userLogin = useSelector((state) => state.userLogin);
  const { userInfo } = userLogin;

  const updateToken = useSelector((state) => state.updateToken);
  const { loading, auth } = updateToken;

  const banHandler = async () => {
    const config = {
      headers: {
          'Content-type': 'application/json',
          Authorization: `Bearer ${auth.access}`   
      }
    } 
    const response = await axios.put(`http://insightwearai.sytes.net:8000/api/users/ban/${author_id}`, {}, config)
    window.location.reload()
  }

  const dismissReportHandler = async () => {
    const config = {
      headers: {
          'Content-type': 'application/json',
          Authorization: `Bearer ${auth.access}`   
      }
    } 
    const response = await axios.delete(`http://insightwearai.sytes.net:8000/api/posts/report/dismiss/${id}`, config)
    window.location.reload()
  }

  const deletePostHandler = async () => {
    const response = await axios.delete(`http://insightwearai.sytes.net:8000/api/post/delete/${post_id}`)
    window.location.reload()
  }

  const viewAuthor = () => {
    navigate('/profileScreen', { state: {data: author_id} })
  }

  const viewSender = () => {
    navigate('/profileScreen', { state: {data: sender_id} })
  }

  const { id, sender_id, sender_name, sender_email, post_id, post_content, post_title, message, author_id, author_name } = props.report

  return (
    <>
      <Container className="adminCardContainer">
          <div className="adminCardBody">
            <Row>
              <Col xs={8} md={10}>
                <Row className="adminCardRow">
                  <h4>Post Author: {author_name}</h4>
                </Row>
                <Row className="adminCardRow">
                  <h4>{post_title}</h4>
                </Row>
                <Row className="adminCardRow">
                  <p>{post_content}</p>
                </Row>
                <Row className="adminCardRow">
                  <p><strong>{sender_name}</strong> ({sender_email}): "{message}"</p>
                </Row>
                <Row className="adminCardRow">
                  <hr style={{width: "100%"}}/>
                </Row>
              </Col>
              <Col xs={4} md={2} style={{display:'flex', justifyContent:'right'}}>
                <Row>
                  <DropdownButton id="dropdownButton" title=''>
                    <Dropdown.Item as="button" onClick={viewAuthor}>View Author</Dropdown.Item>
                    <Dropdown.Item as="button" onClick={viewSender}>View Sender</Dropdown.Item>
                    <Dropdown.Item as="button" onClick={dismissReportHandler}>Dismiss Report</Dropdown.Item>
                    <Dropdown.Item as="button" onClick={deletePostHandler}>Delete Post</Dropdown.Item>
                    <Dropdown.Item as="button" onClick={banHandler}>Ban User</Dropdown.Item>
                  </DropdownButton>
                </Row>
              </Col>
            </Row>
          </div>
      </Container>
    </>
  );
};

export default AdminPostCard;
