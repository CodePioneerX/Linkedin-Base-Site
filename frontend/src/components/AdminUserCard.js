import React, { useState, useEffect } from "react";
import { useSelector } from "react-redux";
import { useNavigate } from 'react-router-dom';
import axios from "axios";
import { Card, Container, Button, Row, Col, Dropdown, DropdownButton } from 'react-bootstrap';
import { Link } from "react-router-dom";
import { faGripVertical } from "@fortawesome/free-solid-svg-icons";
import '../Assets/css/Admin.css';

const AdminUserCard = (props) => {
    
  const navigate = useNavigate();

  // Define state variables
  const [profile, setProfile] = useState("");
  const [reportMessages, setReportMessages] = useState("");

  const userLogin = useSelector((state) => state.userLogin);
  const { userInfo } = userLogin;

  const updateToken = useSelector((state) => state.updateToken);
  const { loading, auth } = updateToken;

// Function to get user profile information
  const getProfile = async () => {
    // Send GET request to retrieve user profile information
    const {data} = await axios.get(
      `http://insightwearai.sytes.net:8000/api/profile/${props.userId}`
    );
    // Update state with retrieved profile information
    setProfile(data.profile);
  };

  const getReportMessages = async () => {
    const config = {
      headers: {
          'Content-type': 'application/json',
          Authorization: `Bearer ${auth.access}`   
      }
    } 

    // Send GET request to retrieve user report messages
    const {data} = await axios.get(
      `http://insightwearai.sytes.net:8000/api/users/reports/${props.userId}`, config
    );
    // Update state with retrieved report messages
    setReportMessages(data);
  }

  const banHandler = async () => {
    const config = {
      headers: {
          'Content-type': 'application/json',
          Authorization: `Bearer ${auth.access}`   
      }
    }
    const response = await axios.put(`http://insightwearai.sytes.net:8000/api/users/ban/${props.userId}`, {}, config)
    window.location.reload()
  }

  const dismissReportHandler = async () => {
    const config = {
      headers: {
          'Content-type': 'application/json',
          Authorization: `Bearer ${auth.access}`   
      }
    }
    const response = await axios.delete(`http://insightwearai.sytes.net:8000/api/users/report/dismiss/${props.userId}`, config)
    window.location.reload()
  }

  const viewProfile = () => {
    navigate('/profileScreen', { state: {data: props.userId} })
  }

  // Call getProfile() function when component mounts
  useEffect(() => {
    getProfile();
    getReportMessages();
  }, []);

  return (
    <>
      <Container className="adminCardContainer">
        <div className="adminCardBody">
          <Row>
            <Col xs={8} md={10}>
              <Row className="adminCardRow">
                <img className="img-fluid rounded-pill adminCardProfileImage" src={profile.image}></img>
                <h4>{profile.name}</h4>
              </Row>
              <Row className="adminCardRow">
                <hr style={{width: "100%"}}/>
              </Row>
              {reportMessages && reportMessages.length > 0 && reportMessages.map(report => (
                <Row className="adminCardRow" key={report.id}>
                  <p><strong>{report.sender_name}</strong> ({report.sender_email}): "{report.message}"</p>
                </Row>
              ))
              }
              <Row className="adminCardRow">
                <hr style={{width: "100%"}}/>
              </Row>
            </Col>
            <Col xs={4} md={2} style={{display:'flex', justifyContent:'right'}}>
              <Row>
                <DropdownButton id='dropdownButton' title=''>
                  <Dropdown.Item as="button" onClick={viewProfile}>View Profile</Dropdown.Item>
                  <Dropdown.Item as="button" onClick={dismissReportHandler}>Dismiss Report</Dropdown.Item>
                  <Dropdown.Item as="button" onClick={banHandler}>Ban User</Dropdown.Item>
                </DropdownButton>
              </Row>
            </Col>
          </Row>
        </div>
      </Container>
    </>
  );
};

export default AdminUserCard;
