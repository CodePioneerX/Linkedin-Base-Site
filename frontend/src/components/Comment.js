// Importing necessary packages from 'mdb-react-ui-kit' and 'react'
import React from 'react';
import {
  MDBCard,
  MDBCardBody,
  MDBCardImage,
  MDBCol,
  MDBContainer,
  MDBIcon,
  MDBInput,
  MDBRow,
} from "mdb-react-ui-kit";

// Functional component Comment with props 'author', 'content', 'createdAt'
const Comment = ({ author, content, createdAt }) => {
  return (
    // Using MDBContainer and MDBRow to create a container and row to hold the comment
    <MDBContainer>
        <MDBRow> 
    // Using MDBCard from MDBReact to create a card for the comment
        <MDBCard className="mb-4">
                <MDBCardBody>
                  // Adding a prompt for users to add their note
                  <p>Type your note, and hit enter to add it</p>
                  
                  // Dividing the card into two parts - left and right
                  <div className="d-flex justify-content-between">
    
                    // Left part with the author's image and name
                    <div className="d-flex flex-row align-items-center">
                      <MDBCardImage
                        src="https://mdbcdn.b-cdn.net/img/Photos/Avatars/img%20(4).webp"
                        alt="avatar"
                        width="25"
                        height="25"
                      />
                      <p className="small mb-0 ms-2">Martha</p>
                    </div>
    
                     // Right part with upvote count and an upvote icon
                    <div className="d-flex flex-row align-items-center">
                      <p className="small text-muted mb-0">Upvote?</p>
                      <MDBIcon
                        far
                        icon="thumbs-up mx-2 fa-xs text-black"
                        style={{ marginTop: "-0.16rem" }}
                      />
                      <p className="small text-muted mb-0">3</p>
                    </div>
                  </div>
                </MDBCardBody>
              </MDBCard>
        </MDBRow>
    </MDBContainer>
  );
};

export default Comment;
