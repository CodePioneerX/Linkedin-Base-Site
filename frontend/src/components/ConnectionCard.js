import React, { useState, useEffect } from "react";
import axios from "axios";
import { Card, Button, Row, Col } from 'react-bootstrap';
import { Link } from "react-router-dom";
import { faGripVertical } from "@fortawesome/free-solid-svg-icons";

const ConnectionCard = (props) => {

// const shortAbout = about.split(".")[0] + '...';
const [profile, setProfile] = useState("");

const getProfile = async () => {
    var userId;

    switch(props.type) {
      case 'received':
        userId = props.senderId;
        break;
      case 'sent':
        userId = props.recipientId;
        break;
      case 'possible':
        userId = props.recipientId;
        break;
      default:
        userId = props.recipientId;
        break;
    }

    const {data} = await axios.get(
      `http://insightwearai.sytes.net:8000/api/profile/${userId}`
    );
    setProfile(data.profile);
    
  };

  const acceptHandler = async () => {
    // accept connection request
    const response = await axios.put(
      `http://insightwearai.sytes.net:8000/api/connections/accept/${props.senderId}/${props.recipientId}/`
    )
    window.location.reload()
  }

  const rejectHandler = async () => {
    // reject connection request
    const response = await axios.delete(
      `http://insightwearai.sytes.net:8000/api/connections/reject/${props.senderId}/${props.recipientId}/`
    )
    window.location.reload()
  }

  const cancelHandler = async () => {
    // cancel sent connection request
    const response = await axios.delete(
      `http://insightwearai.sytes.net:8000/api/connections/cancel/${props.senderId}/${props.recipientId}/`  
    )
    window.location.reload()
  }

  const sendConnectionRequestHandler = async () => {
    // send connection request
    const response = await axios.post(
      `http://insightwearai.sytes.net:8000/api/connections/create/${props.senderId}/${props.recipientId}/`  
    )
    window.location.reload()
  }

  useEffect(() => {
     getProfile(); 
  }, []);

  return (
    <>
      <Card >
        <Row>
          <Card.Body className='card_body'style={{ height:'90px'}} >
            <div style={{display:'flex', justifyContent:'left'}}>
              <Col style={{display:'flex', justifyContent:'left'}}>
                  <Card.Img className='img-fluid rounded-pill' 
                  style={{width:'50px'}}
                  src={profile.image} />  
                  <Card.Title style={{marginLeft:'1rem',marginTop:'.7rem'}}>{profile.name}</Card.Title>             
              </Col>
              <Col style={{display:'flex', justifyContent:'right'}}>
              {props.status == 'accepted' ? 
                <Link to="/profileScreen" state={{data:props.senderId}}>
                  <Button variant="primary">View Profile</Button>
                </Link> : 
                <>
                  {props.type == 'received' && 
                    <>
                      <Link to="/profileScreen" state={{data:props.senderId}}>
                        <Button variant="primary">View Profile</Button>
                      </Link>
                      <div style={{paddingRight: "10px"}}></div>  
                      <Button style={{height: '38px'}} onClick={acceptHandler} variant="secondary">Accept</Button>
                      <div style={{paddingRight: "10px"}}></div>  
                      <Button style={{height: '38px'}} onClick={rejectHandler} variant="danger">Reject</Button>
                    </>
                  }
                  {props.type == 'sent' && 
                    <>
                      <Link to="/profileScreen" state={{data:props.recipientId}}>
                          <Button variant="primary">View Profile</Button>
                      </Link>
                      <div style={{paddingRight: "10px"}}></div>  
                      <Button style={{height: '38px'}} onClick={cancelHandler} variant="danger">Cancel</Button>
                    </> 
                  }
                  {props.type == 'possible' && 
                    <>
                      <Link to="/profileScreen" state={{data:props.recipientId}}>
                        <Button variant="primary">View Profile</Button>
                      </Link>
                      <div style={{paddingRight: "10px"}}></div>  
                      <Button style={{height: '38px'}} onClick={sendConnectionRequestHandler} variant="secondary">Send Connection Request</Button>
                    </> 
                  }
                </>
              }
              </Col>
            </div>
          </Card.Body>
        </Row>
      </Card>
    </>
  );
};

export default ConnectionCard;
