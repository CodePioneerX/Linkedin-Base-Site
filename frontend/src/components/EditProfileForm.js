
import Button from 'react-bootstrap/Button';
import React, { useState, useEffect} from 'react'
import { useDispatch, useSelector } from "react-redux";
import '../Assets/css/Login.css';
import {  Form, FormGroup, Label, Input, Row, Col } from 'reactstrap';
import axios from 'axios';
import { update_profile } from '../actions/userActions'

export const EditProfileForm =(profile)=>{
    const [name, setName] = useState(profile.profile.name)
    const [title, setTitle] = useState(profile.profile.title)
    const [city, setCity] = useState(profile.profile.city)
    const [about, setAbout] = useState(profile.profile.about)
    const [experience, setExperience] = useState(profile.profile.experience)
    const [education, setEducation] = useState(profile.profile.education)
    const [image, setImage] = useState(profile.profile.image)
    const [imagePath, setImagePath] = useState(profile.profile.image)
    const [work, setWork] = useState(profile.profile.work)
    const [volunteering, setVolunteering] = useState(profile.profile.volunteering)
    const [courses, setCourses] = useState(profile.profile.courses)
    const [projects, setProjects] = useState(profile.profile.projects)
    const [awards, setAwards] = useState(profile.profile.awards)
    const [languages, setLanguages] = useState(profile.profile.languages)

    const userLogin = useSelector((state) => state.userLogin);
    const { userInfo } = userLogin;
    
    const dispatch = useDispatch();
    
    const submitHandler = (e)=>{
        e.preventDefault()
        if (name ===""){
            alert("Name cannot be empty!")
            return;
        }
        dispatch(update_profile(userInfo.id, name, title, city, about, experience, education, image, work, volunteering, courses, projects, awards, languages))
        
        window.location.reload(false);
    }

    const cancelHandler = (e) => {
        e.preventDefault()
        window.location.reload(false)
    }
    
    return <div>
        <h2>Edit Your Profile</h2>
        <Form >
        <FormGroup className='mb-4'>
        <Label className='labelE' for="name" >User Name</Label>
            <Input  name="name" value = {name} id="name" placeholder={profile.profile.name} 
            onChange={(e)=> setName(e.target.value)}/>
        </FormGroup>
        <FormGroup className='mb-4'>
        <Label className='labelE' for="title">User Title</Label>
            <Input  name="title" value = {title} id="title" placeholder={profile.profile.title} 
            onChange={(e)=> setTitle(e.target.value)}/>
        </FormGroup>
        <FormGroup className='mb-4'>
        <Label className='labelE' for="city">City</Label>
            <Input  name="city" value = {city} id="city" placeholder={profile.profile.city} 
            onChange={(e)=> setCity(e.target.value)}/>
        </FormGroup>
        <FormGroup className='mb-4'>
        <Label className='labelE' for="about">About me</Label>
            <textarea  className= 'form-control' 
            name="about" id="about" value = {about} placeholder={profile.profile.about} 
            onChange={(e)=> setAbout(e.target.value)}/>
        </FormGroup>
        <FormGroup className='mb-4'>
        <Label className='labelE' for="experience">Experience</Label>
            <textarea className= 'form-control'
            name="experience" id="experience" value= {experience} placeholder={profile.profile.experience} 
            onChange={(e)=> setExperience(e.target.value)}/>
        </FormGroup>
        <FormGroup className='mb-4'>
        <Label className='labelE' for="education">Education</Label>
            <textarea  className= 'form-control'
            name="education" id="education" value = {education} placeholder={profile.profile.education} 
            onChange={(e)=> setEducation(e.target.value)}/>
        </FormGroup>
        <FormGroup className='mb-4'>
        <Label className='labelE' for="work">Work</Label>
            <textarea  className= 'form-control'
            name="work" id="work" value = {work} placeholder={profile.profile.work} 
            onChange={(e)=> setWork(e.target.value)}/>
        </FormGroup>
        <FormGroup>
        <Label className='labelE' for="volunteering">Volunteering</Label>
            <textarea  className= 'form-control'
            name="volunteering" id="volunteering" value = {volunteering} placeholder={profile.profile.volunteering} 
            onChange={(e)=> setVolunteering(e.target.value)}/>
        </FormGroup>
        <FormGroup>
        <Label className='labelE' for="courses">Courses</Label>
            <textarea  className= 'form-control'
            name="courses" id="courses" value = {courses} placeholder={profile.profile.courses} 
            onChange={(e)=> setCourses(e.target.value)}/>
        </FormGroup>
        <FormGroup>
        <Label className='labelE' for="projects">Projects</Label>
            <textarea  className= 'form-control'
            name="projects" id="projects" value = {projects} placeholder={profile.profile.projects} 
            onChange={(e)=> setProjects(e.target.value)}/>
        </FormGroup>
        <FormGroup>
        <Label className='labelE' for="awards">Awards</Label>
            <textarea  className= 'form-control'
            name="awards" id="awards" value = {awards} placeholder={profile.profile.awards} 
            onChange={(e)=> setAwards(e.target.value)}/>
        </FormGroup>
        <FormGroup>
        <Label className='labelE' for="languages">Languages</Label>
            <textarea  className= 'form-control'
            name="languages" id="languages" value = {languages} placeholder={profile.profile.languages} 
            onChange={(e)=> setLanguages(e.target.value)}/>
        </FormGroup>

       
        <FormGroup className='mb-4'>
        <div className="profile-header">
              <p className='labelE'>Your profile image now</p>
              <img src={profile.profile.image} className='editImg' />
        </div>
        </FormGroup>
        <FormGroup className='mb-4'>
        <Label className='labelE ' for="image">Choose your new profile image</Label>
            <input type='file' name="image" id="image" 
            onChange={(e)=> {setImage(e.target.files[0])
                            setImagePath('/images/'+ e.target.files[0].name)}}/>
        </FormGroup>
     
        <Row className='editButtonContainer'>
                <Col xs={12} md={6}>
                    <Button type = 'submit' className='editSaveButton' onClick={submitHandler}> Save </Button>
                </Col>
                <Col xs={12} md={6}>
                    <Button className='editCancelButton' onClick={cancelHandler}>Cancel</Button>
                </Col>
            </Row>
        </Form>
    </div>
}
