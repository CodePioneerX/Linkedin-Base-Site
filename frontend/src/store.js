import { createStore, combineReducers, applyMiddleware} from 'redux'
import thunk from 'redux-thunk'
import { composeWithDevTools } from 'redux-devtools-extension'
import { userRegisterReducer, 
        userLoginReducer, 
        userProfileReducer } from './reducers/userReducers'

        

const reducer = combineReducers({
    userLogin: userLoginReducer,
    userRegister: userRegisterReducer,
    userProfile: userProfileReducer,
  
})

const userInfoFromStorage = localStorage.getItem("userInfo") ? 
        JSON.parse(localStorage.getItem("userInfo")) : null

        
const initialState = {
    userLogin: { userInfo: userInfoFromStorage },
 
    userProfile: { }
}
const middleware = [thunk]

const store = createStore(reducer, initialState, 
    composeWithDevTools(applyMiddleware(...middleware)))

export default store;