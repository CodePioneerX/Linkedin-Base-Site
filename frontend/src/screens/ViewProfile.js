
// Importing required dependencies and components from libraries and files

import React, { useState, useEffect } from "react";
import axios from "axios";
import { Container, Row, Col } from "react-bootstrap";
import { useDispatch, useSelector } from "react-redux";
import Posts from "../components/Posts-old";
import { useNavigate, Link } from "react-router-dom";
import Alert from "react-bootstrap/Alert";
import Tab from 'react-bootstrap/Tab';
import Tabs from 'react-bootstrap/Tabs';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faPenToSquare } from '@fortawesome/free-regular-svg-icons';
import { EditProfileForm } from '../components/EditProfileForm';
import { EditPostForm } from '../components/EditPostForm';
import '../Assets/css/App.css'
import { get_profile } from '../actions/userActions'
import ConnectionCard from '../components/ConnectionCard';
import RecommendationCard from "../components/RecommendationCard";


function ViewProfile() {
  
  // Setting initial state variables using React's `useState` hook
  
  const [profile, setProfile] = useState("");
  const [recommendation, setRecommendation] = useState("");    //Holds and sets the value
  const [receivedRecommendations, setReceivedRec]= useState(""); 
  const [editor, setEditor] = useState(false)
  const [postEditor, setPostEditor] = useState(false)
  
  const [post, setPost] = useState('') // Holds and sets the value of user's post data

  const dispatch = useDispatch();
  const userLogin = useSelector((state) => state.userLogin);
  const { userInfo } = userLogin;
  const navigate = useNavigate();

  
  
 // If user is not logged in, redirect to the login page, else call the `getProfile` function
  
  
  useEffect(() => {
    if (!userInfo) {
      navigate("/");
    } else {
    getProfile();
  }
  }, [userInfo, navigate]);
  
  
  

  const getProfile = async () => {
    const { data } = await axios.get(
      `http://insightwearai.sytes.net:8000/api/profile/` + userInfo.id
    );
    setProfile(data.profile);
    setRecommendation(data.sent_recommendations);
    // console.log(data)
    setReceivedRec(data.received_recommendations);

    // console.log(data)
  };

  
  
   // If user is not logged in, redirect to the login page, else call the `getProfile` function
  useEffect(() => {
    getProfile(); //getProfile function call 
    // dispatch(get_profile(userInfo.id))
  }, []);

  
  
  
  const editorMode = ()=>{
    setEditor(true)
  }
 
  const quitEditor = ()=>{
    setEditor(false)
  }

  const postEditorMode = () => {
    setPostEditor(true)
  }

  const quitPostEditor = () => {
    setPostEditor(false)
  }

  const createPost = () => {
    navigate('/create/post/')
  }
  
  
  

  return (
    
    <Container className="justify-content-md-center padd">
    
    
    {/* check if user info is available */}
      {userInfo ? ( 
        <div className="profile-page">
    
    {/*Some  styling */} 
          <div style={{ display: "flex" }}>
            <div style={{ flex: 5}}>
            {editor ? <EditProfileForm profile={profile} quitEditor={quitEditor}/> : 
            postEditor ? <EditPostForm post={post} quitPostEditor={quitPostEditor}/> : 
              <Container>
                <Row>
                  {/* Bio + Resume Column  */}
                  <Col sm={12} md={12} lg={8}>
                    {/* <Row> */}
                    <div className="profile-main-card">
                      <Row style={{backgroundImage: `url(${require("../images/a.jpg")})`, paddingBottom: "15px", backgroundSize: "cover", backgroundPosition: "center center", display: "flex", flexDirection: "column", alignItems: "center"}}>     
                        <img src={profile.image} alt="Profile" className="profile-image padd_small" style={{ borderRadius: "50%", width: "auto", height: "250px"}}/>
                      </Row>

                      <Row style={{background:"white",display: "flex", flexDirection: "column", alignItems: "center"}}>
                        <div style={{textAlign: "center", padding: "1rem"}}>
                          <h1 className="profile-name">{profile.name}</h1>
                          <h4 className="profile-title">{profile.title}</h4>
                          <h6 className="profile-city" style={{paddingBottom:"10px"}}>{profile.city}</h6>
                          <button className="profile-button">Connections</button>
                          <button className="profile-button" onClick={editorMode}>Edit Profile</button>
                          <button className="profile-button">Contact Info</button>
                        </div>
                      </Row>
                    </div>
                      <Row>
                        <Col>
                          <div className="profile-header">
                            <div className="profile-card">
                              <h2 className="padd_small"><b>About Me</b></h2>
                              <p className="about-me">{profile.about}</p>
                            </div>
                            <div className="profile-card">
                              <h2 className="padd_small"><b>Experience</b></h2>
                              <ul className="experience-list">{profile.experience}</ul>
                            </div>

                            <div className="profile-card">
                              <h2 className="padd_small"><b>Education</b></h2>
                              <ul className="education-list">{profile.education}</ul>
                            </div>

                            <div className="profile-card">
                              <h2 className="padd_small"><b>Recommendations</b></h2>
                              
                     
                              <Tabs style={{paddingTop:"1rem"}}
                                defaultActiveKey="recieved"
                                id="receivedRecommendations"
                                className="mb-3">
                  
                              <Tab eventKey="received" title="Received">
                                {receivedRecommendations && Array.isArray(receivedRecommendations) && receivedRecommendations.map((rec, index) => (
                                  <RecommendationCard key={rec.id} senderId={rec.sender} recipientId={rec.recipient} description={rec.description} type={'received'}/>
                                ))}
                              </Tab>
                              <Tab eventKey="recommended" title="Recommended">
                                {recommendation && Array.isArray(recommendation) && recommendation.map((rec, index) => (
                                  <RecommendationCard key={rec.id} senderId={rec.sender} recipientId={rec.recipient} description={rec.description} type={'sent'}/> 
                                ))}
                              </Tab>
                            </Tabs>             
                            </div>

                            <div className="profile-card">
                              <h2 className="padd_small"><b>Work</b></h2>
                              <ul className="work-list">{profile.work}</ul>
                            </div>

                            <div className="profile-card">
                              <h2 className="padd_small"><b>Volunteering</b></h2>
                              <ul className="volunteering-list">{profile.volunteering}</ul>
                            </div>

                            <div className="profile-card">
                              <h2 className="padd_small"><b>Courses</b></h2>
                              <ul className="courses-list">{profile.courses}</ul>
                            </div>

                            <div className="profile-card">
                              <h2 className="padd_small"><b>Projects</b></h2>
                              <ul className="projects-list">{profile.projects}</ul>
                            </div>

                            <div className="profile-card">
                              <h2 className="padd_small"><b>Awards</b></h2>
                              <ul className="awards-list">{profile.awards}</ul>
                            </div>

                            <div className="profile-card">
                              <h2 className="padd_small"><b>Languages</b></h2>
                              <ul className="languages-list">{profile.languages}</ul>
                            </div>

                          </div>
                        </Col>
                      </Row>
                  </Col>
                  
                  {/* Activity Column */}
                  <Col sm={12} md={12} lg={4}>
                    <Row>
                        <div className="profile-header" style={{ backgroundColor: "white", borderRadius: "7px", borderBottom: "1px solid #ccc", boxShadow: "0 0 5px rgba(0, 0, 0, 0.1)", padding: "1.5rem"}}>
                          <Row>
                            <Col md={8}><h2>Activities</h2></Col>
                            <Col md={4}>
                              <Link className='profile-button' to='/create/post/' state={{from: "/profile/"}}>
                                <button style={{ backgroundColor: "#3D13C6", color: "white", borderRadius: "25px", padding: "5px 10px", border: "none" }}><FontAwesomeIcon icon={faPenToSquare} style={{ color: "white"}}/> </button>
                              </Link>
                            </Col>
                          </Row>
                          <Posts u_id={userLogin.userInfo.id} edit={postEditorMode} quit={quitPostEditor} setpost={setPost}/>
                        </div>
                    </Row>
                  </Col>
                </Row>
              </Container>
      }
        </div>        
      </div>
      <Row>
        <div className="profile-content">
          <Container className="justify-content-md-center pd-5 padd_small"></Container>
        </div>
      </Row>
    </div>
      ) : (
        <Alert className="alertLogin" key="primary" variant="primary">
          <h5>
            You are not signed in! Please <a href="/login">Sign in</a> or{" "}
            <a href="/register">Register</a>!
          </h5>
        </Alert>
      )}
    </Container>
  );
}
export default ViewProfile;
