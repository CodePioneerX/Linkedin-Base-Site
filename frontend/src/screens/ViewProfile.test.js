import React from 'react';
import { shallow, configure} from 'enzyme';
import { render, screen, waitFor } from '@testing-library/react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'; // import the adapter
import { useDispatch, useSelector } from "react-redux";
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import ViewProfile from './ViewProfile';
import { EditProfileForm } from '../components/EditProfileForm';
import store from '../store';

jest.mock('react-redux', () => ({
    useSelector: jest.fn(),
    useDispatch: jest.fn(),
  }));

const mockedUsedNavigate = jest.fn();
  jest.mock('react-router-dom', () => ({
     ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedUsedNavigate,
  }));

configure({ adapter: new Adapter() });

describe('LoginPage', () => {
    let wrapper;
    beforeEach(() => {
        const userInfo = { id: 1 };
        
        useSelector.mockReturnValue({ userInfo });
        wrapper = shallow(
        // <Provider store={store}>
        // <Router>
          <ViewProfile/>,{ editor: true }
    //     </Router>
    //    </Provider> 
        
        );
      });
  
    it('should render profile page', async () => {
        expect(wrapper).toMatchSnapshot();
  });

  it('should render the EditProfileForm component when Editor is true', () => {
    //wrapper.setState({ jobEditor: true });
    const profile = { name:"test name",
                      title: "test title"
};
    wrapper = shallow(<EditProfileForm profile={profile}/>,{ editor: true });
    expect(wrapper.find(EditProfileForm)).toHaveLength(0);
  });

  

  

});