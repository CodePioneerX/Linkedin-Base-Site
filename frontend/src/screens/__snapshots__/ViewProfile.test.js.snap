// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage should render profile page 1`] = `ShallowWrapper {}`;
